api-definitions.ts

```diff
@@ -0,0 +1,336 @@
+/**  
+    Actions to support in MVP:
+    read board ✅
+    read cards ✅
+    rename board ✅
+    create column ✅
+    rename column ✅
+    move column ✅
+    delete column ✅
+    create card ✅
+    move card ✅
+    delete card ✅
+
+    Actions to support later: 
+    Move cards between boards
+    create board
+    delete board
+
+    Other thoughts:
+        - we have to support cards being added/removed from other places (task overview)
+        - we have to support cards being added/removed from legacy (tasks and lessons)
+        - we might want to load board information from rough to detailed (we had the idea to first render loading states for all cards, and then query each card individually)
+        - create and delete board is not necessarily needed in the first version
+        - single requests for moving single elements, or reorder?
+
+*/
+
+/*
+  Regarding Multi-Tenant conflicts: 
+
+  Util we have Websockts to keep all clients (almost) in sync with each other, we have to deal with Conflicts in some way.
+
+  Option 1: We track a Board-Version for every mutation to the board and return a `409 - Conflict` on version mismatch. 
+            When there is a mismatch the board gets refreshed.
+
+  Option 2: We track a Board-Version for each board mutation. When there is a Version mismatch greater than +1 
+            (e.g. last known version is 10, backend returns a 12) the Frontend reloads the board.
+
+
+  **Intermediary decision**
+  Option 3: We ignore multi-tenant edits until we have websockts. 
+            On each board-mutation, we return the full board-skeleton and update to minimize potential conflicts.			
+*/
+
+interface CourseBoardTaskCard {}
+
+interface CouseBoardContentCard {}
+
+interface CourseBoardElement {}
+
+type CourseBoardCardType = "task" | "content" | "legacy-task" | "legacy-lesson";
+
+// GET - Load a CourseBoard
+// /boards/:id
+
+interface CourseBoard {
+	id: string;
+	title: string;
+	columns: CourseBoardColumn[];
+	version: number;
+	timestamps: CourseBoardTimestamps;
+}
+
+interface CourseBoardTimestamps {
+	lastUpdatedAt: string;
+	createdAt: string;
+	deletedAt: string;
+}
+
+interface CourseBoardColumn {
+	id: string;
+	title: string;
+	cards: CourseBoard_Skeleton_Card[]; // ?
+}
+
+interface CourseBoard_Skeleton_Card {
```

Comment:
Why the underscores?

Reply:
the name **Skeleton**Card sparked some discussion that we skipped in favor of making progress.

the underscores are there to indicate that. 

I think Skeleton is a well name describing the purpose of the interface.

```suggestion
interface CourseBoardSkeletonCard {
```